plugins/out_stackdriver/stackdriver.c

```diff
@@ -761,21 +761,67 @@ static int get_stream(msgpack_object_map map)
     return STREAM_UNKNOWN;
 }
 
+static int make_bool_map(struct flb_stackdriver *ctx, msgpack_object *map,
+                         bool_map_t *bool_map, int map_num, 
+                         flb_sds_t to_be_removed[], int removed_num) {
+    msgpack_object_kv *kv;
+    msgpack_object *key;
+
+    char is_to_delete;
+    int i, j;
+    int ret = 0;
+    flb_sds_t removed;
+
+    for (i = 0; i < map_num; i++) {
+        bool_map[i] = TO_BE_REMAINED;
+    }
+
+    /* tail of map */
+    bool_map[map_num] = TAIL_OF_ARRAY;
+    if (map != NULL) {
+        kv = map->via.map.ptr;
+        for (i = 0; i < map_num; i++) {
+            key = &(kv + i)->key;
+            is_to_delete = FLB_FALSE;
+            if (key->type == MSGPACK_OBJECT_STR) {
+                for (j = 0; j < removed_num; j++) {
+                    removed = to_be_removed[j];
+                    if (key->via.str.size != flb_sds_len(removed)) {
+                        continue;
+                    }
+
+                    if (strncasecmp(key->via.str.ptr, removed,
+                                    key->via.str.size) == 0) {
+                        is_to_delete = FLB_TRUE;
+                        break;
+                    }
+                }
+
+                if (is_to_delete == FLB_TRUE) {
+                    bool_map[i] = TO_BE_REMOVED;
+                    ret++;
+                }
+            }
+        }
+    }
+
+    /* return number of elements removed from payLoad */
+    return ret;
+}
+
 static int pack_json_payload(int operation_extracted, int operation_extra_size, 
                              msgpack_packer* mp_pck, msgpack_object *obj,
                              struct flb_stackdriver *ctx)
 {
     /* Specified fields include local_resource_id, operation, sourceLocation ... */
-    int i, j;
+    int i;
     int to_remove = 0;
     int ret;
     int map_size;
     int new_map_size;
-    int len;
     int len_to_be_removed;
-    int key_not_found;
-    flb_sds_t removed;
     flb_sds_t local_resource_id_key;
+    bool_map_t bool_map[128];
```

Comment:
if map_size > 128 we will get a really bad overflow. Please allocate bool_map from heap using the proper map size from the record.

Reply:
updated